src/jllvm/materialization/JNIImplementationLayer.cpp

```diff
@@ -130,80 +124,104 @@ void jllvm::JNIImplementationLayer::emit(std::unique_ptr<llvm::orc::Materializat
 
                 llvm::IRBuilder<> builder(llvm::BasicBlock::Create(*context, "entry", function));
 
-                llvm::Value* environment = builder.CreateAlloca(llvm::StructType::get(builder.getPtrTy()));
-                builder.CreateStore(
-                    builder.CreateIntToPtr(builder.getInt64(reinterpret_cast<std::uintptr_t>(m_jniNativeFunctions)),
-                                           builder.getPtrTy()),
-                    environment);
-
-                builder.CreateCall(module->getOrInsertFunction("jllvm_push_local_frame", builder.getVoidTy()));
+                llvm::Type* returnType = descriptorToType(methodType.returnType(), *context);
+                llvm::Type* referenceType = jllvm::referenceType(*context);
```

Comment:
```suggestion
                llvm::Type* referenceType = referenceType(*context);
```

Reply:
This does not work because of the symbol names.